release-notes/cabal-install-3.14.2.0.md

```diff
@@ -0,0 +1,103 @@
+
+  Local+noindex repositories will have their `noindex.cache` file created the first time they are accessed.
+
+- Fix regression where 'build' folder was created in CWD when running a standlone script. [#10772](https://github.com/haskell/cabal/issues/10772) [#10800](https://github.com/haskell/cabal/pull/10800)
+
+  Fix a regression where the `build` folder was created in the current directory
+  when running a standalone script.
+
+- Fix duplicate environment variables in test and benchmark runs [#10718](https://github.com/haskell/cabal/issues/10718) [#10827](https://github.com/haskell/cabal/pull/10827)
+
+  Cabal no longer creates duplicate environment variables when running test
+  suites, benchmarks, or internal executables. Previously, when setting up the
+  environment for these processes, Cabal would append the overridden environment
+  to the existing environment, creating duplicates of the same variable.
+
+- Deduplicate "using configuration from" message [#10546](https://github.com/haskell/cabal/pull/10546)
+
+  Deduplicates and sorts the list of configuration files and URIs printed with the
+  "using configuration from" message. This message is shown when there's a build
+  failure. We can trigger that message by using a non-existant package in the
+  project, "no-pkg-dir".
+
+  If an import is repeated in a `.project` or `.config` file it only imported once
+  but if the same import is made from an imported file then it was being repeated
+  in the message. Additional problems were not showing the project first and
+  mixing configuration files and URIs together.
+
+  * Before the fix:
+
+    ```
+    $ ~/.ghcup/bin/cabal-3.12.1.0 build all --dry-run
+    When using configuration from:
+    - a-very-extra.config
+    - an-extra.config
+    - cabal.project
+    - https://www.stackage.org/lts-21.25/cabal.config
+    - https://www.stackage.org/lts-21.25/cabal.config
+    - https://www.stackage.org/lts-21.25/cabal.config
+    - z-empty.config
+    The following errors occurred:
+    - The package location 'no-pkg-dir' does not exist.
+    ```
+
+  * After the fix:
+
+    ```
+    $ cabal build all --dry-run
+    When using configuration from:
+    - cabal.project
+    - a-very-extra.config
+    - an-extra.config
+    - z-empty.config
+    - https://www.stackage.org/lts-21.25/cabal.config
+    The following errors occurred:
+    - The package location 'no-pkg-dir' does not exist.
+      ```
+- Add a custom `Ord` instance for `ProjectConfigPath` ([#10546](https://github.com/haskell/cabal/pull/10546))
+
+  The instance sorts URIs after local
+  file paths and longer file paths after shorter ones as measured by the number of
+  path segments. If still equal, then sorting is lexical.  The project itself, a
+  single element root path, compared to any of the configuration paths it imports,
+  should always sort first. Comparing one project root path against another is
+  done lexically.
+
+### Unresolved Issues
```

Comment:
Thinking aloud: would it better to put it as the first section? At least as long as it's short? That's worse publicity, but better transparency.

Reply:
It's a reasonable idea but it's a little strange to start with it, I feel. If we had a proper distinction between changelog and release notes, we could allow more prose in the latter, including an opening paragraph that could advertise this section. But since it's just a changelog, I don't even know that "unresolved issues" make any sense here. I think @Kleidukos' idea with this section was to put it in the announcement that we post on Discourse and elsewhere (even the Haskell Blog https://blog.haskell.org/), rather than the changelog.

Note that the document is called "release-notes.md" but it's really meant to be a changelog (and is redirected to from the actual changelog file), so I expect no prose and only a list of actual changes here.

Reply:
OK, so maybe it's less mysterious and alarming at the end. Nothing prevents use from adding a few sentences of prose at the start, BTW, and we sometimes do if there's anything unusual to warn the users about, but to mandate it would make releases even longer and even less obvious. I'm dropping this suggestion.